### PR TITLE
[Security] $attributes can be anything, but RoleVoter assumes strings

### PR DESCRIPTION
**Pull request by maennchen**
| Q | A |
| --- | --- |
| Branch? | 3.0 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18042 |
| License | MIT |
| Doc PR | reference to the documentation PR, if any |

This is the merged to 3.0 version of #19725.

### CODE REVIEW

File: src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php [hunk 1]
```diff
@@ -41,13 +41,19 @@ public function vote(TokenInterface $token, $subject, array $attributes)
         $roles = $this->extractRoles($token);
 
         foreach ($attributes as $attribute) {
+            if (is_array($attribute)) {
+                continue;
+            }
+            if (is_object($attribute) && !method_exists($attribute, '__toString')) {
+                continue;
+            }
```
Comment (ro0NL) [inline src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php]: Statements can be removed. Imo we should not silently ignore invalid values by design.

**Comment by lemoinem** [comment]
Hello, Thank you but unless there are merge conflicts, your PR on 2.7 will be automatically merged in every supported branches, including 3.0, 3.1 and master...

From what I can see, the patch are completely equivalent, so I don't think you need this PR.

Thank you for the hard work!

**Comment by maennchen** [comment]
@lemoinem Since the method `supportsAttribute` does not exist anymore in newer branches, there will be a merge conflict. This would be the solution to that merge conflict.

**Comment by lemoinem** [comment]
Ok, I see the difference now.
Status: Reviewed (Dependent on #19725, of course)

**Comment by lemoinem** [comment]
Status: Reviewed

**Comment by nicolas-grekas** [comment]
Can you rebase+change the base branch to 3.1 so that tests are against 3.1? 3.0 is not maintained anymore.

**Comment by maennchen** [comment]
@nicolas-grekas Won't you merge the changes into `2.7` (#19725) ? Then you will encounter the merge error starting with the `3.0` version. This PR is just to fix the merge error.

If you want to only apply starting from `2.7`, leave a gap and restart at `3.1`, I will rebase it.

**Pull request closed without merging**